paper-server/patches/sources/net/minecraft/world/level/ServerExplosion.java.patch

```diff
@@ -234,6 +207,22 @@
          this.level.gameEvent(this.source, GameEvent.EXPLODE, this.center);
          List<BlockPos> list = this.calculateExplodedPositions();
          this.hurtEntities();
+@@ -254,6 +_,15 @@
+             this.interactWithBlocks(list);
+             profilerFiller.pop();
+         }
++        // Paper start - handle not interacts in explosion events
++        else {
++            io.papermc.paper.event.world.ExplodeEvent explodeEvent = CraftEventFactory.callExplodeEvent(this, list);
++            this.wasCanceled = explodeEvent.isCancelled();
++            this.yield = explodeEvent.getYield();
```

Comment:
The yield shouldn't be needed there no blocks explode anyway

Reply:
yeah but not make "damage" keep that.